src/ergtable.h

```diff
@@ -0,0 +1,140 @@
+#ifndef ERGTABLE_H
+#define ERGTABLE_H
+
+#include <QList>
+#include <QSettings>
+#include <QObject>
+#include <cmath>
+
+#include <QMetaType>
+#include <QDebug>
+
+class DataPoint {
+public:
+    double cadence = 0; // RPM, default initialized
+    double wattage = 0; // Watts, default initialized
+    double resistance = 0; // Some unit, default initialized
+
+    DataPoint() = default; // Default constructor
+
+    DataPoint(double c, double w, double r) : cadence(c), wattage(w), resistance(r) {} // Custom constructor
+};
+
+Q_DECLARE_METATYPE(DataPoint)
+
+class ergTable : public QObject {
+    Q_OBJECT
+
+public:
+    ergTable(QObject *parent = nullptr) : QObject(parent) {
+        loadSettings();
+    }
+
+    ~ergTable() {
+        saveSettings();
+    }
+
+    void collectData(double cadence, double wattage, double resistance) {
```

Comment:
Something that stands out to me about this approach is that you're keying the data using floating point numbers. 
If I look at this code as in independent component that could be used by anything, rather than the 2 I see currently (fakebike and trxappgateusbbike) I see the potential for every sample collected to create a new record, causing the list to grow, and your estimateWattage function, which scans the whole table every time it's called, would take longer. Significantly longer on modern CPU? Maybe not, but I prefer to write code that doesn't stress an RP Pico.

A solution: bucketing. Key by integer cadence and resistance and round to nearest. Perhaps you're doing it implicitly already because the only values you're getting from the hardware for cadence and resistance are actually integer.

Reply:
Actually I wrote without thinking about performance. For example they can be uint16.

I would like first to make a usable version and then optimize it :)

Reply:
e.g. (60.2, 200.1, 50.5), (60.1, 200.1, 50.7)...

Reply:
now are uint16